triggers/tests/src/test/scala/com/digitalasset/daml/lf/engine/trigger/TriggerRuleSimulationLib.scala

```diff
@@ -66,7 +66,24 @@ private class TriggerRuleMetrics {
     metricTimingData = mutable.Map.empty
   }
 
-  def getMetrics: TriggerRuleMetrics.RuleMetrics = {
+  def getMetrics(
+      retries: Int = 5
+  )(implicit materializer: Materializer): Future[TriggerRuleMetrics.RuleMetrics] = {
+    val backoff = 50.milliseconds
+    val timeLimit = backoff * retries
+    val restartSettings =
+      RestartSettings(minBackoff = backoff, maxBackoff = 1.second, randomFactor = 0.1)
+        .withMaxRestarts(retries, timeLimit)
```

Comment:
On source failure (e.g. `getMetrics` fails assertion checking):
- we restart (and so retry calling `getMetrics`)
- 5 retries over `timeLimit` occur
- each retry is double the previous retry attempt (plus or minus a 10% jitter)